sdks/go/pkg/beam/windowing.go

```diff
@@ -21,21 +21,47 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+type WindowIntoOption interface {
+	windowIntoOption()
+}
+
+type WindowTrigger struct {
+	Name window.TriggerType
+}
+
+func (t WindowTrigger) windowIntoOption() {}
+
+func (t WindowTrigger) GetName() window.TriggerType {
```

Comment:
There's no need for a GetName function. As a rule, don't add "Get" to a method that just returns a field.
See https://golang.org/doc/effective_go#Getters